sbi/inference/snle/snle_base.py

```diff
@@ -167,6 +134,23 @@ def __call__(
 
             x = self._batched_simulator(theta)
 
+            # First round or if retraining from scratch:
+            # Call the `self._build_neural_net` with the rounds' thetas and xs as
+            # arguments, which will build the neural network
+            # This is passed into NeuralPosterior, to create a neural posterior which
+            # can `sample()` and `log_prob()`. The network is accessible via `.net`.
+            if round_ == 0 or retrain_from_scratch_each_round:
+                self._posterior = NeuralPosterior(
+                    method_family="snle",
+                    neural_net=self._build_neural_net(theta, x),
```

Comment:
we could move the check for `NaN`s up to happen just after `x` is simulated. Otherwise `NaN`s might influence the z-scoring transform, or another check inside `_build_neural_net` would be required, no?

Reply:
see other comment below.